### PR TITLE
doc: update README.md to notice the user about RisingWave's usage of Scarf

### PR DESCRIPTION
**Pull request by alicelyy**
A few examples of other open source projects using similar communication: 
- https://github.com/apache/superset/wiki/Superset-Third%E2%80%90Party-Dependencies
- https://github.com/ngx-lottie/ngx-lottie
- https://github.com/apache/sedona/blob/c878485ef7f5941baa2871d6f0cc03b782e41a5c/docs/asf/telemetry.md?plain=1
- https://github.com/search?q=%22https%3A%2F%2Fscarf.sh%2F%22+language%3AMarkdown&type=code&l=Markdown

Our telemetry communication: 
https://docs.risingwave.com/docs/current/telemetry/#how-to-opt-out

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]

**Comment by xiangjinwu** [comment]
There is a **Notes on telemetry** section below already.

**Comment by alicelyy** [comment]
> There is a **Notes on telemetry** section below already.
> 
> > RisingWave collects anonymous usage statistics to better understand how the community is using RisingWave. The sole intention of this exercise is to help improve the product. Users may opt out easily at any time. Please refer to the [user documentation](https://docs.risingwave.com/docs/current/telemetry/) for more details.

Telemetry is not our only usage of Scarf. We also track user IP from docker download link (pending CNAME change from our network infra provider) as well as various website pages visits. For this reason I think it may be more appropriate to put the notice at a higher level location.

File: README.md [hunk 1]
```diff
@@ -71,6 +71,9 @@ Then follow the prompts to start and connect to RisingWave.
 
 To learn about other installation options, such as using a Docker image, see [Quick Start](https://docs.risingwave.com/docs/current/get-started/).
 
+> Please note: RisingWave uses [Scarf](https://scarf.sh/) to collect anonymized installation analytics. These analytics help support us understand and improve the distribution of our package. However, if you'd like to opt out, you can do so by setting scarf.enabled = false in your project's risingwave/src/config/<your-config>.toml.
```
Comment (hengm3467) [inline README.md:74]:
```suggestion
> **Note**: RisingWave uses [Scarf](https://scarf.sh/) to collect anonymized installation analytics. These analytics help us understand and improve the distribution of our package. If you prefer to opt out, you can do so by setting `scarf.enabled = false` in your project's `risingwave/src/config/<your-config>.toml` file.
```

Comment (hengm3467) [inline README.md:74]:
```suggestion
> **Note**: RisingWave uses [Scarf](https://scarf.sh/) to collect anonymized installation analytics. These analytics help us understand and improve the distribution of our package. If you prefer to opt out, you can do so by setting `scarf.enabled = false` in your project's `risingwave/src/config/<your-config>.toml` file.
```

**Pull request merged**